staff/sergio-luz/pre-course/pr3-bingo/project-T3.js

```diff
@@ -0,0 +1,225 @@
+
+//-VARIABLES-//
+var UserName=[
+	{name:"J1",points:50}, 
+	{name:"J2",points:80}
+]
+var TamañoCarton=15;
+var CartonK=[];
+var ListaBolas=[];
+var Puntos;
+var comprobarLinea=true;
+//-//
+function BINGO(){
+	var Turno=0, comprobarLinea=true;
+	Puntos=100;
+	var name=PedirNombre();
+
+	UserName.push({name:name, points:Puntos});
+	alert("Bienvenido "+name);
+	InformarFuncionamientoPuntos();
+
+	AskTurn(Turno);
+}
+//-//
+function PedirNombre(){	//Cuando se ejecute, pedirá el nombre del jugador y deberá guardarse. 
+	var x="";
+	while(x=="")	{x=prompt("Escribe tu nombre porfavor:", "Eustaquio");}
+	console.log("Bienvenido "+x+". Empieza a jugar:");
+	return x;
+}
+//-//
+function InformarFuncionamientoPuntos(){
+	confirm("Todos los jugadores empiezan inicialmente con 100 puntos. \nPor cada bola que salga perderán uno.\nEn caso de abandonar el juego no se almacenarán los puntos.\nCompleta tu carton más rapido que los demás para ganar.")
+	console.log("Todos los jugadores empiezan inicialmente con 100 puntos. \nPor cada bola que salga perderán uno.\nEn caso de abandonar el juego no se almacenarán los puntos.\nCompleta tu carton más rapido que los demás para ganar.")
+}
+//-//
+function AskTurn(turno){
+	var Completed=CartonCompletado();
+	if(turno==0){
+		//Primer turno
+		BorrarCarton()
+		CreateCarton(TamañoCarton)
+	}else if(Completed==true){
+		FinishGame();
+	}else{
+		Esperar();
+		ConfirmNewTurn();
+	}
+}
+//-//
+function CartonCompletado(){
+	var i=0;
+	for(i=0;i<CartonK.length;i++){		
+		if(CartonK[i].matched==false){
+			if(comprobarLinea==true){
+				LineaCompletada();
+			}
+			return false;
+		}
+	}
+	return true;
+}
+function LineaCompletada(){
+	//Para cada linea
+	for(var i=1; i<4; i++){
+		//Obtenemos un array con los numeros de esa linea
+		var TemporalArray=CartonK.filter(R=> (R.linea==i));
+		//Comprobamos si en esa linea existe algún número NO tachado.
+		var fallos=TemporalArray.filter(R=> (R.matched==false));
+		//En caso de no fallar ha habido una linea
+		if(fallos==0){
+			console.log("¡¡¡LINEA!!!")
+			comprobarLinea=false;
+			break;
+		}
+	}
+}
+//-//
+function BorrarCarton(){
+
+	CartonK=[]; //vaciamos el antiguo carton
+}
+function CreateCarton(T){	// Durante el primer turno se mostrará un cartón con 15 números (excluyendo el 0 siempre)
+	//Construimos el carton
+	var i=0;
+	for(i=0;i<T;i++){		
+		var x= GenerarRandom();
+		//comprobamos si el número ya existe en el cartón
+		if(CartonK.some(elem => elem.number ==x)){
+			CreateCarton(TamañoCarton-(CartonK.length));
+			return;
+		}
+		var casilla={number:x,matched:false};
+		CartonK.push(casilla);
+	}
+	//Los ordenamos
+	CartonK.sort(function (a, b){ return (a.number - b.number) }); //Menor a mayor
+	//Creamos las 3 lineas
+	for(var i=0; i<15; i++){
+		if(i<5){
+			CartonK[i].linea=1;
+		}else if(i<10){
+			CartonK[i].linea=2;
+		}else if(i<15){
+			CartonK[i].linea=3;
+		}
+	}
+	// Mostramos el carton
+	ShowCarton()
+	//Preguntamos si quiere este carton
```

Comment:
do not use differents lenguages for function name.

Reply:
done